src/main/java/com/griddynamics/cloud/learning/dal/repositories/Repository.java

```diff
@@ -0,0 +1,78 @@
+package com.griddynamics.cloud.learning.dal.repositories;
+
+import com.griddynamics.cloud.learning.dal.AbortedQueryException;
+import com.griddynamics.cloud.learning.dal.connection.DBConnectionManager;
+import com.griddynamics.cloud.learning.dal.domain.DomainEntity;
+import com.griddynamics.cloud.learning.dal.dto.DomainEntitiesDTO;
+import com.griddynamics.cloud.learning.dal.dto.ArticleRequestParamsDTO;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public abstract class Repository<T extends DomainEntity> {
+
+    private static final Logger logger = Logger.getLogger(Repository.class.getName());
+
+    protected abstract String getEntityTableName();
+
+    protected abstract Optional<T> convertToEntity(ResultSet resultSet);
+
+    public Optional<T> getById(final long id) {
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE id = " + id + ";";
+        return getEntityByParameter(query);
+    }
+
+    protected Optional<T> getEntityByParameter(final String query) {
+        try (Connection connection = DBConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.next();
+            return convertToEntity(resultSet);
+        } catch (SQLException e) {
+            final String msg = "Unable to retrieve required entity. Query \"" + query + "\" aborted.";
+            logger.log(Level.SEVERE, msg, e);
+            throw new AbortedQueryException(msg, e);
+        }
+    }
+
+    protected DomainEntitiesDTO<T> getEntitiesDTO(final String query, final ArticleRequestParamsDTO params) {
```

Comment:
You are using only two fields from the dto, it's getting not very obvious why do we need this dto at all.
It'd be better to have just pageNumber and limit in the method's signature.